pkg/db/tx.go

```diff
@@ -128,18 +120,28 @@ func (d *dbTx) Exec(query string, args ...interface{}) (pgconn.CommandTag, error
 		"query": queryToString(query),
 		"took":  time.Since(start),
 	})
-	if isUniqueViolation(err) {
-		return nil, ErrAlreadyExists
-	}
 	if err != nil {
-		dbErrorsCounter.WithLabelValues("exec").Inc()
-		log.WithError(err).Error("SQL query failed with error")
-		return res, err
+		return res, d.handleSQLError(err, "exec", query)
 	}
 	log.Trace("SQL query executed successfully")
 	return res, err
 }
 
+func (d *dbTx) handleSQLError(err error, cmdType string, query string) error {
+	dbErrorsCounter.WithLabelValues(cmdType).Inc()
+	d.logger.WithError(err).Error("SQL query failed with error")
```

Comment:
The main concern I have in with this one, maybe it can be addressed outside this pr - we log error level and count for all errors.
Before this change, In case we 'select' all repositories and didn't get any result - now for this case, we log an error and increase the error metric.
Also, additional logging after this log, like in like 139 will just fill the logs with additional information which is already known.
Verify that I understood the code right, but if this is the case, we may want to move these lines after the 'exists' and 'not found' issues.

Reply:
The concern is that the original code condition some specific errors (isUniqueViolation for example).
The previous version didn't log as error and didn't count the error metric in this case.

Reply:
Got it!